include/dali/core/byte_io.h

```diff
@@ -21,7 +21,10 @@ namespace dali {
 
 namespace detail {
 
-template <int nbytes, bool is_little_endian, typename T>
+template <int nbytes,
+          bool is_little_endian,
+          typename T,
+          std::enable_if_t<std::is_integral<T>::value, bool> = true>
 void ReadValueImpl(T &value, const uint8_t* data) {
```

Comment:
This usage of `enable_if` is typical for constructors/destructors. Here, you can go the usual way:
```suggestion
template <int nbytes, bool is_little_endian, typename T>
std::enable_if_t<std::is_integral<T>::value> ReadValueImpl(T &value, const uint8_t* data) {
```